mlflow/pyfunc/model.py

```diff
@@ -361,5 +382,14 @@ def _convert_input(self, model_input):
 
         return model_input
 
-    def predict(self, model_input):
-        return self.python_model.predict(self.context, self._convert_input(model_input))
+    def predict(self, model_input, params: Optional[Dict[str, Any]] = None):
+        """
+        :param model_input: Model input data.
+        :param params: Additional parameters to pass to the model for inference.
+
+                       .. Note:: Experimental: This parameter may change or be removed in a future
+                                               release without warning.
+
+        :return: Model predictions.
+        """
+        return self.python_model.predict(self.context, self._convert_input(model_input), params)
```

Comment:
@serena-ruan I tried running the following custom pyfunc logging / inference code, which works in MLflow 2.4:

```
import mlflow

class MyMod(mlflow.pyfunc.PythonModel):
    def predict(self, ctx, inp):
        return 7


with mlflow.start_run():
    m = mlflow.pyfunc.log_model(
        python_model=MyMod(),
        artifact_path="foo"
    )

m = mlflow.pyfunc.load_model(m.model_uri)
print(m.predict(["k"]))
```

This throws on the PR branch with the following exception:

```
Traceback (most recent call last):
  File "log.py", line 17, in <module>
    print(m.predict(["k"]))
  File "/Users/corey.zumar/mlflow/mlflow/pyfunc/__init__.py", line 442, in predict
    return self._predict_fn(data, params)
  File "/Users/corey.zumar/mlflow/mlflow/pyfunc/model.py", line 395, in predict
    return self.python_model.predict(self.context, self._convert_input(model_input), params)
TypeError: predict() takes 3 positional arguments but 4 were given
```

Can we fix this and add a test case for compatibility? We shouldn't break existing workflows that define custom pyfunc models